packages/apptive_grid_core/lib/src/network/apptive_grid_client.dart

```diff
@@ -14,18 +14,28 @@ import 'package:openid_client/openid_client.dart';
 class ApptiveGridClient {
   /// Creates an ApiClient
   ApptiveGridClient({
-    this.options = const ApptiveGridOptions(),
+    ApptiveGridOptions options = const ApptiveGridOptions(),
     http.Client? httpClient,
     ApptiveGridAuthenticator? authenticator,
-  }) : _client = httpClient ?? http.Client() {
+  })  : _options = options,
+        _client = httpClient ?? http.Client() {
     this.authenticator = authenticator ??
-        ApptiveGridAuthenticator(options: options, httpClient: _client);
+        ApptiveGridAuthenticator(client: this, httpClient: _client);
     _attachmentProcessor =
         AttachmentProcessor(options, this.authenticator, httpClient: _client);
   }
 
   /// Configurations
-  ApptiveGridOptions options;
+  ApptiveGridOptions _options;
+
+  /// Returns the [ApptiveGridOptions] for the current Client
+  ApptiveGridOptions get options => _options;
+
+  set options(ApptiveGridOptions options) {
+    // Keep old environment as change is handled in [updateEnvironment]
+    _options = options.copyWith(environment: _options.environment);
+    updateEnvironment(options.environment);
```

Comment:
Could this being asynchronous be a problem? Would it be possible to change it into an async function?